### PR TITLE
Airlock Seals 2: I'm dumb and a fool edition

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
This is basically just an exact remake of #52595 except without all the bumbling fuckups I made due to being dumb. Mega thanks to Rohesie for helping me, considering literally all of the problems were ones I made for myself (the merge conflict was literally due to a missing cricket bat icon). In conclusion, I'm dumb.

> Adds door seals to the game, inspired by the ones used in Alien Isolation and SOMA. Trivially easy for a human to remove, but makes it a nightmare to try and smash through if you're an alien, an evil robot, or don't have hands. I also slightly changed how airlock reinforcement works, as now there are two ways to increase the integrity, the previous way of just resetting the max_integrity after the removal of the plating wouldn't work.

## Why It's Good For The Game
Makes it harder for aliens, cyborgs, and simplemobs to get into places the crew don't want them, while being easy to remove by humans.

## Changelog
:cl:
add: Adds door seals to the game, printable at engineering and security lathes after they've been researched. Makes airlocks much harder to smash down if you can't remove them.
tweak: Slightly changed how bonus health to an airlock works
/:cl:

### CODE REVIEW

**Review by Twaticus — APPROVED** [review]

**Comment by actioninja** [comment]
I forgot to ever comment again but yeh I actually just completely misunderstood the feature, it actually seems fine.

**Comment by cacogen** [comment]
This will make me play Alien: Isolation finally. Fuck aliens and fuck simple mobs.

**Comment by anconfuzedrock** [comment]
wait a second aren't airlock seals basically reinforced tables
gotta be human to climb a table

**Comment by Jimmius** [comment]
sort of, a lot of the things this is supposed to counter can just walk over tables anyway, like spiders and xenos and what have you. It also doubles the integrity of the airlock, which welding doesn't do.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -70,6 +71,7 @@
 	var/obj/item/electronics/airlock/electronics
 	COOLDOWN_DECLARE(shockCooldown)
 	var/obj/item/note //Any papers pinned to the airlock
+	var/obj/item/seal //The seal on the airlock
```
Comment (Timberpoes) [inline code/game/machinery/doors/airlock.dm:74]: This needs to be documented as per our [documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -34,6 +34,7 @@
 
 #define AIRLOCK_INTEGRITY_N			 300 // Normal airlock integrity
 #define AIRLOCK_INTEGRITY_MULTIPLIER 1.5 // How much reinforced doors health increases
+#define AIRLOCK_SEAL_MULTIPLIER		 2 // How much extra health airlocks get when braced with a seal
```
Comment (Timberpoes) [inline code/game/machinery/doors/airlock.dm:37]: This needs to be documented as per our [documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -97,6 +98,10 @@
 			return TRUE
 	return FALSE
 
+/// Triggered when attempting to remove an airlock seal
+/obj/machinery/door/proc/try_remove_seal(mob/user)
+	return
```
Comment (Timberpoes) [inline code/game/machinery/doors/door.dm:103]: Could use improved documentation as per our [documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

Please add information on the user argument for try_remove_seal. Doesn't have to be long, as long as it's documented as per the guidelines.

**Review by Timberpoes — CHANGES_REQUESTED** [review]

**Comment by Timberpoes** [comment]
Geez laweez. Sorry this has been hanging so long. I've now assigned myself to it.

I'll see about getting this merged as soon as. It's definitely overdue and I'm not sure how you slipped under our radars.

Current triage is as follows:

- [ ] Code review -> Minor notes requesting improved documentation.
- [ ] Icon review -> I'll try and poke @Twaticus or someone else to review the new icons and sprite work.
- [ ] Have you tested this? -> While the above is happening, you can help expedite the process by explaining any testing you've done to be sure this feature works as intended.
- [ ] Mergies -> When the above 3 steps are completed, we'll get this in.

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (Jimmius) [reply code/game/machinery/doors/airlock.dm]: I  think  that's resolved now

File: code/game/machinery/doors/door.dm (hunk 3 above)
Comment (Jimmius) [reply code/game/machinery/doors/door.dm]: I think that's resolved

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (Jimmius) [reply code/game/machinery/doors/airlock.dm]: I think this is resolved now

**Comment by Jimmius** [comment]
Things  wot I have tested so far:

1: I have used  the seal on every single  airlock currently  in the game. It works and displays correctly, and I can remove it from all of them. 
2: Non-Cultists can safely remove this from Cult airlocks without getting bounced, so culties don't get a free lunch.
3: If an airlock with safety unlocks is unpowered and sealed, clicking will remove the seal (you can unlock the safety system once it's removed).
4: The seal keeps the door closed during emags and staff of doors, and blocks attempts to open from the AI and Cyborgs, as well as pulsing the open wire.
5: Everything works fine if you seal AND reinforce and airlock, in either order.

If there's any other edge case interactions that I can't think of, just say and I'll test them.

File: code/game/machinery/doors/door.dm [hunk 4]
```diff
@@ -98,7 +98,14 @@
 			return TRUE
 	return FALSE
 
-/// Triggered when attempting to remove an airlock seal
+/**
+  * Called when attempting to remove the seal from an airlock
+  *
+  * Here because we need to call it and return if there was a seal so we don't try to open the door
+  * or try it's safety lock while it's sealed
```
Comment (Timberpoes) [inline code/game/machinery/doors/door.dm:105]:
```suggestion
  * or try its safety lock while it's sealed
```

Comment (Timberpoes) [inline code/game/machinery/doors/door.dm:105]:
```suggestion
  * or try its safety lock while it's sealed
```

**Review by Timberpoes — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/door.dm (hunk 4 above)
Comment (Timberpoes) [reply code/game/machinery/doors/door.dm:105]: Yes, I am being that pedantic! Muahahaha.

He, his. Her, hers. It, its.

Comment (Jimmius) [reply code/game/machinery/doors/door.dm:105]: fuck, how did I not notice this

**Review by Timberpoes — APPROVED** [review]
